src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -78,10 +79,15 @@ func (r *StatefulSetResource) runUpdate(
 	if err = r.updateRestartingStatus(ctx, true); err != nil {
 		return fmt.Errorf("unable to turn on restarting status in cluster custom resource: %w", err)
 	}
+
 	if err = r.updateStatefulSet(ctx, current, modified); err != nil {
 		return err
 	}
 
+	if err = r.isClusterHealthy(ctx); err != nil {
```

Comment:
i'm not too familiar with this part of the controller and i am tired from flight :sweat_smile: please forgive if this is stupid q, but i thought we implement our own rolling update process. should this check be after that?

Reply:
It is inside our rolling update process. The https://github.com/redpanda-data/redpanda/blob/2d775e6c35ae37279be2a5cc18ba29b5ebb78c1e/src/go/k8s/pkg/resources/statefulset_update.go#L43-L59